anta/tests/routing/bgp.py

```diff
@@ -277,3 +362,323 @@ def test(self) -> None:
+    """
+
+    name = "VerifyBGPPeersHealth"
+    description = "Verifies the health of BGP peers"
+    categories = ["routing", "bgp"]
+    commands = [
+        AntaTemplate(template="show bgp {afi} {safi} summary vrf {vrf}"),
+        AntaTemplate(template="show bgp {afi} summary"),
+    ]
+
+    class Input(AntaTest.Input):  # pylint: disable=missing-class-docstring
+        address_families: List[BgpAfi]
+        """
+        List of BGP address families (BgpAfi)
+        """
+
+        class BgpAfi(BaseModel):  # pylint: disable=missing-class-docstring
+            afi: Afi
+            """BGP address family (AFI)"""
+            safi: Optional[Safi] = None
+            """Optional BGP subsequent service family (SAFI).
+
+            If the input `afi` is `ipv4` or `ipv6`, a valid `safi` must be provided.
+            """
+            vrf: str = "default"
+            """
+            Optional VRF for IPv4 and IPv6. If not provided, it defaults to `default`.
+
+            If the input `afi` is not `ipv4` or `ipv6`, e.g. `evpn`, `vrf` must be `default`.
+            """
+
+            @model_validator(mode="after")
+            def validate_inputs(self: BaseModel) -> BaseModel:
+                """
+                Validate the inputs provided to the BgpAfi class.
+
+                If afi is either ipv4 or ipv6, safi must be provided.
+
+                If afi is not ipv4 or ipv6, safi must not be provided and vrf must be default.
+                """
+                if self.afi in ["ipv4", "ipv6"]:
+                    if self.safi is None:
+                        raise ValueError("'safi' must be provided when afi is ipv4 or ipv6")
+                elif self.safi is not None:
+                    raise ValueError("'safi' must not be provided when afi is not ipv4 or ipv6")
+                elif self.vrf != "default":
+                    raise ValueError("'vrf' must be default when afi is not ipv4 or ipv6")
+                return self
+
+    def render(self, template: AntaTemplate) -> list[AntaCommand]:
+        commands = []
+        for afi in self.inputs.address_families:
+            if template == VerifyBGPPeersHealth.commands[0] and afi.afi in ["ipv4", "ipv6"]:
+                commands.append(template.render(afi=afi.afi, safi=afi.safi, vrf=afi.vrf))
+            elif template == VerifyBGPPeersHealth.commands[1] and afi.afi not in ["ipv4", "ipv6"]:
+                commands.append(template.render(afi=afi.afi, vrf=afi.vrf))
+        return commands
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        self.result.is_success()
+
+        failures: dict[tuple[str, Any], dict[str, Any]] = {}
+
+        for command in self.instance_commands:
+            if command.params:
+                command_output = command.json_output
+
+                afi = cast(Afi, command.params.get("afi"))
+                safi = cast(Optional[Safi], command.params.get("safi"))
+                afi_vrf = cast(str, command.params.get("vrf"))
```

Comment:
Addin a comment here to track an issue to see if we can pass `Input` to params when rendering a template rather than a dict loosing the types